### ISSUE TITLE
invalid escape sequence in regular expression

### ISSUE THREAD

**Issue by technobuddha** [bellard/quickjs#374]
quickjs compilation is failing with the error message `invalid escape sequence in regular expression` on the following code:

```
const re = /[\-]/u;
```

However, if the 'u' flag is omitted it compiles fine.

I found this problem using the very popular npm library `ajv-formats` (26 million DL/week)

**Comment by bnoordhuis** [comment]
I fixed that in 2023 in quickjs-ng/quickjs@f0ef9e159387105da55907fbc8d58a4d1fe920b4 although more as a side effect of a larger change. I think it should be possible to cherry-pick only the relevant bits though.

**Comment by benatkin** [comment]
It's more of a [feature addition](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/unicodeSets#browser_compatibility) than a fix IMO

**Comment by technobuddha** [comment]
> It's more of a [feature addition](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/unicode#browser_compatibility) than a fix IMO

I don't see how this is a "feature".   It's clearly a bug.   Node, and most browsers accept the regex syntax.   The syntax is part of the regexp documentation.

**Comment by benatkin** [comment]
I understand it better now and yeah it isn't a feature. It's a bug that only shows up when the code has an unnecessarily escaped `-` in a unicode string.

**Comment by technobuddha** [comment]
@benatkin Exactly.   I would consider this low-priority except for the fact that this is in several very popular npm packages including ajv.

**Comment by bellard** [comment]
fixed
